### PR TITLE
NH-65068 Add is_lambda checks and todos to APM Python

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
This adds `is_lambda` in-code checks and distro setup to APM Python. Also adds TODOs for what's next for lambda mode that I hope are self-explanatory. Big pieces to note:

### (1) _get_extension_components

The base of this approach is a little draconian: new function `ApmConfig._get_extension_components` declares which parts of the c-lib extension should be no-op depending on `agent_enabled` and `is_lambda`. Right now it's:

```
agent_enabled T, is_lambda F -> init all c-lib including Reporter
agent_enabled T, is_lambda T -> all no-op
agent_enabled F              -> all no-op
```

After introducing c-lib settings API usage update in https://swicloud.atlassian.net/browse/NH-64716, it _will_ be

```
agent_enabled T, is_lambda F -> init all c-lib including Reporter EXCEPT OboeAPI
agent_enabled T, is_lambda T -> init ONLY c-lib OboeAPI
agent_enabled F              -> all no-op
```

I'm doing it this way because we do not want any non-OboeAPI business triggered by APM Python when in lambda, based on testing with Jerry.

### (2) _calculate_agent_enabled_config_lambda

New function `ApmConfig._calculate_agent_enabled_config_lambda` is separate from the config check if outside lambda. I'm thinking the only user-side config that would disable APM Python inside lambda would be if `SW_APM_AGENT_ENABLED` is false. As per testing c-lib updates with Jerry, no need for service key.

### (3) calculate_liboboe_decision

Updates existing function `calculate_liboboe_decision` with a condition if `is_lambda`. It drops trace at the moment until c-lib settings API usage update in https://swicloud.atlassian.net/browse/NH-64716

----

Please let me know what you think!

### CODE REVIEW

File: solarwinds_apm/apm_config.py [hunk 1]
```diff
@@ -267,14 +307,16 @@ def _calculate_agent_enabled_config(self) -> bool:
...
+        #      https://swicloud.atlassian.net/browse/NH-65713
         try:
             # SolarWindsDistro._configure does setdefault so this shouldn't
             # be None, but safer and more explicit this way
             environ_exporters = os.environ.get(
                 OTEL_TRACES_EXPORTER,
                 INTL_SWO_DEFAULT_TRACES_EXPORTER,
             ).split(",")
-            # If not using the default propagators,
+            # If not using the default exporters,
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/apm_config.py:319]: Just fixing an old typo

File: solarwinds_apm/apm_config.py [hunk 2]
```diff
@@ -530,7 +572,6 @@ def _snake_to_camel_case(key):
             self.agent_enabled = cnf_agent_enabled
 
         available_cnf = set(self.__config.keys())
-        # TODO after alpha: is_lambda
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/apm_config.py:533]: As per Slack chat starting [here](https://swicloud.slack.com/archives/C05R52Y2GHK/p1698706088610519), we won't configure `token_*` in APM Python like we did in AO. So no longer needed.

File: solarwinds_apm/apm_config.py [hunk 3]
```diff
@@ -611,11 +652,12 @@ def update_with_env_var(self) -> None:
             self.agent_enabled = env_agent_enabled
 
         available_envvs = set(self.__config.keys())
-        # TODO after alpha: is_lambda
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/apm_config.py:614]: See [above comment](https://github.com/solarwindscloud/solarwinds-apm-python/pull/211/files#r1378243559).

File: solarwinds_apm/apm_config.py [hunk 4]
```diff
@@ -700,7 +742,6 @@ def _set_config_value(self, keys_str: str, val: Any) -> Any:
                 )
                 self.__config[key] = oboe_trigger_trace
             elif keys == ["reporter"]:
-                # TODO: support 'lambda'
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/apm_config.py:703]: See [above comment](https://github.com/solarwindscloud/solarwinds-apm-python/pull/211/files#r1378243559).

File: solarwinds_apm/configurator.py [hunk 5]
```diff
@@ -46,7 +47,6 @@
 )
 from solarwinds_apm.apm_fwkv_manager import SolarWindsFrameworkKvManager
 from solarwinds_apm.apm_meter_manager import SolarWindsMeterManager
-from solarwinds_apm.apm_noop import Reporter
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/configurator.py:49]: This was actually the wrong import used for type checking before. Removed this and added `solarwinds_apm.extension.oboe.Reporter` below.

File: solarwinds_apm/configurator.py [hunk 6]
```diff
@@ -165,12 +170,16 @@ def _configure_sampler(
             ),
         )
 
-    def _configure_metrics_span_processor(
+    def _configure_inbound_metrics_span_processor(
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/configurator.py:173]: Renaming for clarity

**Review by raphael-theriault-swi — DISMISSED** [review]
Looks good !

**Review by raphael-theriault-swi — APPROVED** [review]

**Pull request merged**